core/chaincode/lifecycle/lifecycle.go

```diff
@@ -515,6 +502,42 @@ func (ef *ExternalFunctions) ApproveChaincodeDefinitionForOrg(chname, ccname str
 	return nil
 }
 
+func (ef *ExternalFunctions) isAttemptToRedefine(ccname, privateName, packageID string, requestedSequence int64, cd *ChaincodeDefinition, orgState ReadWritableState) (bool, error) {
```

Comment:
The error messages and code in this new function suggest that the definition is uncommitted however based on the "attempted to redefine the current committed sequence (%d) for namespace %s" message above it looks like the attempt to redefine can also occur for committed definitions which would be confusing. Is that right?

Reply:
You have two cases:

1. You approved and committed and trying to redefine (requestedSequence == currentSequence)
2. You trying to approve something that was approved before (requestedSequence == currentSequence + 1)

In this case, you have an error message:

line 467: "attempted to redefine the current committed sequence (%d) for namespace %s"

In the second case, you have:

line 480: "attempted to redefine uncommitted sequence (%d) for namespace %s with unchanged content"

Reply:
@jt-nti another case to support is when a new organization might join the channel, so the chaincode is already approved and committed by others organizations in the channel. So now, you need to approve this chaincode for the new organization so its peers can endorse it. Hence, it is necessary to look up from the org's private state whether it already has the chaincode definition approved. If it has, you are trying to change something that was approved already before; otherwise, it's fine since the state is empty, and you need to write your approval into the state.

Reply:
Thanks for the examples @C0rWin - the first case "You approved and committed and trying to redefine" seems to confirm that the following should be updated to remove `uncommitted` in the new `isAttemptToRedefine` function: 
- `uncommittedMetadata`
- `"could not fetch uncommitted definition"`
- `"Attempting to redefine uncommitted definition at sequence %d"`
- `uncommittedParameters`

Reply:
@jt-nti fixed as per your comment.

Reply:
Fab, thanks

Reply:
@jt-nti can you plz take a look, I have also added fixes for linters

Reply:
@denyeart, @mbwhite can you guys please take a look on this PR?